packages/core/src/components/way-textarea/way-textarea.tsx

```diff
@@ -0,0 +1,130 @@
+import { Component, Host, h, Prop, Element, Event, EventEmitter, State } from '@stencil/core';
+import { inheritAttributes } from '../../utils/utils';
+@Component({
+  tag: 'way-textarea',
+  styleUrl: 'way-textarea.scss',
+  shadow: true,
+})
+export class WayTextarea {
+  private inheritedAttributes: { [k: string]: any } = {};
+
+  @Element() el!: HTMLElement;
+
+  /**
+   * The state of the textarea's value attribute.
+   */
+  @State() value: string;
+
+  /**
+   * Specifies what type of textarea to use.
+   */
+  @Prop() type: string | undefined;
+
+  /**
+   * The textarea's name attribute.
+   */
+  @Prop({ reflect: true }) name: string | undefined;
+
+  /**
+   * If `true`, the user cannot interact with the textarea.
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   * Specifies how many characters are allowed.
+   */
+  @Prop() maxlength: number;
+
+  /**
+   * The textarea's label. Alternatively, you can use the label slot.
+   */
+  @Prop() label: string | undefined;
+
+ /**
+   * Specifies how many textarea rows to use.
+   */
+  @Prop() rows: number | undefined;
+
+ /**
+   * If `true`, the textarea should autofocus.
+   */
+  @Prop({ reflect: true }) autofocus = false;
+
+
+  /**
+   * Emitted when the textarea has focus.
+   */
+  @Event() wayFocus!: EventEmitter<void>;
```

Comment:
**issue**: should now be `@Event({ eventName: 'way-focus' })` because this is a more common way of naming events in custom elements and doesn't introduce issues in the Vue wrappers.